### ISSUE TITLE
Odd subfield resolver behavior

### ISSUE THREAD

**Issue by OpenSpacesAndPlaces** [graphql-dotnet/graphql-dotnet#1684]
I'm not sure if I just had not done many to many before or if somehow the Identity tables are slightly  different in definition that what I'd used in the past.

For a query like:
`query {   aspNetUsers_list {  id, email, aspNetUserClaims(ignoreClientWarnings: true){ userId,  claimValue} }}
`
```
"data": {
    "aspNetUsers_list": [
      {
        "id": "371b35e6-e1fa-48ba-a36d-24a40e5a4b57",
        "email": "test@test.com",
        "aspNetUserClaims": [
          {
          }
        ]
      }
    ]
  }
```

Tracing through in Debug mode with source code instead of DLLs I noticed that prior to getting to the ExecutionNode portion of the code - the ResolveFieldContext for the subquery had the definitions for the SubFields of  "userId,  claimValue". But once inside ToValue() of ExecutionNode they are mysteriously missing.

To hack around it I did something like this:
```
        public override object ToValue()
        {
            if (SubFields == null && Result == null)
                return null;

            if ((SubFields == null || SubFields.Count == 0) && Result != null && Result.GetType().GetGenericTypeDefinition() == typeof(Dictionary<,>))
            {
                return Result;
            }
            else
            {
                var fields = new Dictionary<string, object>(SubFields.Count);

                foreach (var kvp in SubFields)
                {
                    var value = kvp.Value.ToValue();

                    if (value == null && kvp.Value.FieldDefinition.ResolvedType is NonNullGraphType)
                    {
                        return null;
                    }

                    fields[kvp.Key] = value;
                }

                return fields;
            }
        }
```

Then all is well.
---

Not sure if there is something like this you've seen before or if there is an obvious miss-step in the Schema.

GraphQL 2.4
GraphQL Authorization 2.1.29

**Comment by sungam3r** [comment]
Have you tried this on the current preview version? Honestly from your message I did not understand what is wrong.

**Comment by OpenSpacesAndPlaces** [comment]
> Have you tried this on the current preview version? 
No.

> Honestly from your message I did not understand what is wrong.
To hopefully clarify - at somepoint in the translation to the Execution Node - the SubField definitions are lost for the SubQuery, but the data is there in the Result. So then it just returns an empty object.

**Comment by sungam3r** [comment]
Are you talking about returning the wrong (empty) data set when it actually not empty?

**Comment by OpenSpacesAndPlaces** [comment]
`Are you talking about returning the wrong (empty) data set when it actually not empty?`
Yeah.

It's doing like:
```
if (SubFields == null)
                return null;
```

Meanwhile "Result" has the Dictionary of data from the dataloader.

**Comment by OpenSpacesAndPlaces** [comment]
This looks to be fixed in 3.0!

**Comment by sungam3r** [comment]
Good. Reopen if it happens again.
